tfjs-backend-webgpu/src/kernels/matmul_small_output_size_webgpu.ts

```diff
@@ -0,0 +1,201 @@
+          // Load one tile of A and B into local memory.
+          mm_Asub1[tileRowA][tileColA] =
+              mm_readA(${aOuterSmall ? 'tileRowA' : 'globalRowA'}, tileColA);
+          mm_Bsub1[tileRowB][tileColB] =
+              mm_readB(tileRowB, ${aOuterSmall ? 'globalColB' : 'tileColB'});
+        }
+      } else {
+        if (tileColA < TileInner) {
+          // Load one tile of A and B into local memory.
+          mm_Asub1[tileRowA][tileColA] =
+              mm_readA(${aOuterSmall ? 'tileRowA' : 'globalRowA'},
+                  t * TileInner + tileColA);
+          mm_Bsub1[tileRowB][tileColB] =
+              mm_readB(t * TileInner + tileRowB,
+                  ${aOuterSmall ? 'globalColB' : 'tileColB'});
+        } else {
+          // Compute acc values for a single thread.
+          for (int k = 0; k < TileInner; k++) {
+            acc += mm_Asub2[tileRowA][k] * mm_Bsub2[k][tileColA-TileInner];
+          }
+        }
+      }
+
+      barrier();
+      if (t != 0) {
+        t++;
+      }
+
+      if (t < numTiles) {
+        if (tileColA < TileInner) {
+          // Load one tile of A and B into local memory.
+          mm_Asub2[tileRowA][tileColA] =
+              mm_readA(${aOuterSmall ? 'tileRowA' : 'globalRowA'},
+                  t * TileInner + tileColA);
+          mm_Bsub2[tileRowB][tileColB] =
+              mm_readB(t * TileInner + tileRowB,
+                  ${aOuterSmall ? 'globalColB' : 'tileColB'});
+        } else {
+          // Compute acc values for a single thread.
+          for (int k = 0; k < TileInner; k++) {
+            acc += mm_Asub1[tileRowA][k] * mm_Bsub1[k][tileColA-TileInner];
+          }
+        }
+      }
+
+      barrier();
+    }
+
+    if (tileColA >= TileInner) {
+      mm_write(${aOuterSmall ? 'tileRowA' : 'globalRowA'},
+          ${aOuterSmall ? 'int(gl_WorkGroupSize.x * gl_WorkGroupID.x) + '
+              : ''}tileColA - TileInner, acc);
+    }
+  }
+  `;
+}
+
+export class MatMulSmallOutputSizeWidthProgram implements WebGPUProgram {
+  outputShape: number[];
+  shaderKey: string;
+  dispatchLayout: {x: number[], y: number[], z: number[]};
+  dispatch: [number, number, number];
+  variableNames = ['A', 'B'];
+  workGroupSize: [number, number, number] = [8, 24, 1];
+  aOuterSmall: boolean;
+  addBias: boolean;
+  activation: backend_util.Activation;
+  hasPreluActivationWeights: boolean;
+
+  constructor(
+      aShape: [number, number, number], bShape: [number, number, number],
+      outputShape: [number, number, number], bias: TensorInfo = null,
+      activation: backend_util.Activation = null,
+      preluActivationWeights: TensorInfo = null) {
+    util.assert(aShape[1] <= 16 || bShape[2] <= 16,
+      () => 'This program can be only used when A width is small.');
+    this.outputShape = outputShape;
+
+    this.dispatchLayout = {x: [2], y: [1], z: [0]};
+    if (aShape[1] <= 16) {
+      this.workGroupSize = aShape[1] <= 8 ? [8, 24, 1] : [16, 32, 1];
```

Comment:
Note that the maximum number of workgroup invocations by webgpu is 256. [16, 32, 1] will exceed it.